ast/compile.go

```diff
@@ -900,6 +903,16 @@ func (c *Compiler) checkRuleConflicts() {
 			if r.Head.RuleKind() == SingleValue && len(node.Children) > 0 {
 				if len(ref) > 1 && !ref[len(ref)-1].IsGround() { // p.q[x] and p.q.s.t => check grandchildren
 					for _, c := range node.Children {
+						if len(c.Values) > 0 {
+							childRules := extractRules(c.Values)
+							for _, childRule := range childRules {
+								childRef := childRule.Ref()
+								if childRule.Head.RuleKind() == SingleValue && !childRef[len(childRef)-1].IsGround() {
+									// The child is a partial object rule, so it's effectively "generating" grandchildren.
+									singleValueConflicts = append(singleValueConflicts, childRule.Ref())
+								}
+							}
+						}
 						if len(c.Children) > 0 {
 							singleValueConflicts = node.flattenChildren()
```

Comment:
Just from looking at the code, it looks like `singleValueConflicts` could be set in line 912, and end up being **overwritten**  here. But I guess the two branches are mutually exclusive?

Reply:
I think they are 🤔 . 
But you have a good point. I'll make this an `append` to be on the safe side. Don't think its worthwhile looking for duplicates, though.